http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/RuntimeRoutes.scala

```diff
@@ -363,3 +439,25 @@ final case class CreateRuntime2Request(labels: LabelMap,
                                        welderDockerImage: Option[ContainerImage],
                                        scopes: Set[String],
                                        customEnvironmentVariables: Map[String, String])
+
+sealed trait UpdateRuntimeConfigRequest extends Product with Serializable {
+  def cloudService: CloudService
+}
+object UpdateRuntimeConfigRequest {
+  final case class GceConfig(updatedMachineType: Option[MachineTypeName], updatedDiskSize: Option[Int])
+      extends UpdateRuntimeConfigRequest {
+    val cloudService: CloudService = CloudService.GCE
+  }
+  final case class DataprocConfig(updatedMasterMachineType: Option[MachineTypeName],
+                                  updatedMasterDiskSize: Option[Int],
+                                  updatedNumberOfWorkers: Option[Int],
+                                  updatedNumberOfPreemptibleWorkers: Option[Int])
+      extends UpdateRuntimeConfigRequest {
+    val cloudService: CloudService = CloudService.Dataproc
+  }
+}
+
+final case class UpdateRuntimeRequest(updatedRuntimeConfig: Option[UpdateRuntimeConfigRequest],
+                                      allowStop: Option[Boolean],
```

Comment:
can we make this `Boolean` instead of `Option[Boolean]`? We're defaulting it to `false` anyway, and having it as `Boolean` makes it a cleaner on `caller` side (we don't have to do `.getOrElse(false)`` everywhere